Fix assign{Ips,Macs}

GeoffreyFrogeye opened the pull request:
One is trivial, the other one is described more in details in its commit.

reo101 commented:
Great catches, thanks!

Could you link a source for the new integer behavior? Is it a `commit`/`PR` in upstream nix (with probably a cherry-pick to `Lix`, or something)?

GeoffreyFrogeye commented:
Sure! Here's the original bug report in Lix: https://git.lix.systems/lix-project/lix/issues/423 , which was then cherry-picked to Nix: https://github.com/NixOS/nix/pull/11188 .

reo101 commented:
That's great, thank you! Could you also link to the upstream `nix` pr (11188) in a comment (or two) above the `subtring` calls, explaining why we're doing the cutoff? (I'd love to have it documented (a bit, at least) in the code itself. I'll swiftly merge afterwards. 😄

reo101 commented:
Maybe we should also wait out to see how the discussion at https://github.com/oddlama/nixos-extra-modules/pull/1#issuecomment-2787517638 goes 😄

spacekitteh commented:
Relevant: https://github.com/NixOS/nix/pull/13001

GeoffreyFrogeye commented:
@spacekitteh Woah, this is very cool to see proper bit shift implementations moving forward already!

It might still be a bit before this lands in stable, so I went ahead with @reo101 's great suggestion and added some comments in the code itself. I now realize you asked to specifically mention the PR, I noted the Lix/Nix versions. Hmm, people can still refer to the changelogs this way, which allows to keep the comments not too verbose either. But just say the word and I'll change it :)

I also added a warning about what still doesn't work as noted in https://github.com/oddlama/nixos-extra-modules/pull/1#issuecomment-2787793365 .
